### ISSUE TITLE
Upgrade from 1.11.6 to 1.12.1 fails (MySQL 8 Innodb cluster)

### ISSUE THREAD

**Issue by mihkelvain** [go-gitea/gitea#12061]
- Gitea version (or commit ref): 1.12.1
- Git version:  1.8.3.1
- Operating system: Centos 7
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x ] Not relevant
- Log gist:

## Description

When trying to start up binary version 1.12.1, it fails with following messages in gitea.log
```

2020/06/26 11:21:34 ...ations/migrations.go:300:Migrate() [I] Migration[126]: Fix topic reposito
ry count
2020/06/26 11:21:34 ...m.io/xorm/core/db.go:277:ExecContext() [I] [SQL] DELETE FROM `repo_topic`
 WHERE `repo_id` NOT IN (SELECT `id` FROM `repository`) [] - 496.336µs
2020/06/26 11:21:34 routers/init.go:77:initDBEngine() [E] ORM engine initialization attempt #9/1
0 failed. Error: migrate: do migrate: Error 3098: The table does not comply with the requirement
s by an external plugin.
2020/06/26 11:21:34 routers/init.go:78:initDBEngine() [I] Backing off for 3 seconds
2020/06/26 11:21:37 routers/init.go:71:initDBEngine() [I] ORM engine initialization attempt #10/
10...
```

Since I run gitea on mysql 8 innodb cluster, then best bet is that gitea schema is not compliant with Innodb cluster requirements. Mainly primary key requirement.
I actually reported missing PK a while ago: https://github.com/go-gitea/gitea/issues/8920

I opened this issue, because I'm not 100% sure DB migration fails due to missing PK.

**Comment by mihkelvain** [comment]
JFYI. I migrated gitea schema away from mysql 8 innodb cluster to single instance 5.6 mysql (because I had this one laying around) and upgrade to 1.12.1 was completed without no issues.

**Comment by zeripath** [comment]
Closed by #12639
